transformation/softprob.go

```diff
@@ -0,0 +1,42 @@
+package transformation
+
+import (
+	"fmt"
+	"math"
+)
+
+type TransformSoftprob struct {
+	NClasses int
+}
+
+func (t *TransformSoftprob) Transform(rawPredictions []float64, outputPredictions []float64, startIndex int) error {
+	if len(rawPredictions) != len(outputPredictions) {
+		return fmt.Errorf("expected len(rawPredictions) = %d (got %d)", t.NClasses, len(rawPredictions))
+	}
+
+	sum := 0.0
+
+	for i, v := range rawPredictions {
+		exp := math.Exp(v)
+		outputPredictions[i] = exp
+		sum += exp
+	}
+
+	for i := range outputPredictions {
+		outputPredictions[i] /= sum
```

Comment:
softprob seems the same that softmax (with help of the SoftmaxFloat64Slice) already did?
Can you just reuse softmax transformation?